src/OSPSuite.Infrastructure.Import/Services/Importer.cs

```diff
@@ -146,6 +155,54 @@ public MappingProblem CheckWhetherAllDataColumnsAreMapped(IReadOnlyList<ColumnIn
                .ToList()
          };
       }
+
+      public IReadOnlyList<DataRepository> DataSourceToDataSets(IDataSource dataSource, IReadOnlyList<MetaDataCategory> metaDataCategories,
+         DataImporterSettings dataImporterSettings, string id)
+      {
+         var dataRepositories = new List<DataRepository>();
+
+         for (var i = 0; i < dataSource.DataSets.SelectMany(ds => ds.Data).Count(); i++)
```

Comment:
Here we are using the iterator of the ParsedDataSets to iterate over ImportedDataSets. This is why this is kept as a for loop.

Reply:
I don't get it
Why. Not


        foreach(var data in dataSource.DataSets.SelectMany(ds => ds.Data)

Or is data not ...data?!?

Reply:
we are using the index to get the ImportedDataSet : 

dataSource.DataSetAt(i). This function is creating the ImportedDataSet from the data.

Reply:
Oh I see. THis looks weird because you are getting an index from the dataSource and then you ask the datasource to do sthg with the index. "Tell don't ask"
Can we have something like "dataSource.DataSets.SelectMany(x => x.DataRepo)) "
If the object are created on the fly, then dataSource.DataSetAt(i) should maybe be dataSource.CreateDataRepoFrom(data)? 

I'll need to get the code on my machine to understand. Not a big deal, but I am always suspicious of for(i loop) with modern constructs

Reply:
closing for now. Let's try to keep that in mind for future implementations